### PR TITLE
[docs] Improve codemod related documentation

### LINKED ISSUE #10870: @mui/x-codemod v6.0.0/data-grid/preset-safe doesn't work with satisfies / Typescript >= v4.9
### Steps to reproduce

Link to live example: (required)

Steps:
1. Make a tsx file with a datagrid
2. Use satisifes on a type in that file
3. Run the code mod

### Current behavior

code mod says unknown node type when it hits the satisfies AST node.

### Expected behavior

codemod runs and transforms the code

### Context

"typescript": "5.0.4"

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```



  System:
    OS: macOS 14.0
    CPU: (10) arm64 Apple M1 Max
    Memory: 313.03 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nodenv/versions/18.12.1/bin/node
    Yarn: 1.22.19 - ~/Library/pnpm/yarn
    npm: 8.5.1 - ~/.nodenv/versions/18.12.1/bin/npm
    pnpm: 6.32.11 - ~/.nodenv/versions/18.12.1/bin/pnpm
  Managers:
    Cargo: 1.63.0 - ~/.cargo/bin/cargo
    Homebrew: 4.1.13 - /opt/homebrew/bin/brew
    pip3: 22.2.2 - ~/.pyenv/shims/pip3
    RubyGems: 3.0.3.1 - /usr/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 15.0.0 - /usr/bin/gcc
    Git: 2.40.0 - /opt/homebrew/bin/git
    Clang: 15.0.0 - /usr/bin/clang
    Curl: 8.1.2 - /usr/bin/curl
  Servers:
    Apache: 2.4.56 - /usr/sbin/apachectl
  Virtualization:
    VirtualBox: 7.0.8 - /usr/local/bin/vboxmanage
  SDKs:
    iOS SDK:
      Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
  IDEs:
    VSCode: 1.83.1 - /opt/homebrew/bin/code
    Vim: 9.0 - /usr/bin/vim
    Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
  Languages:
    Bash: 3.2.57 - /bin/bash
    Go: 1.19.2 - /opt/homebrew/bin/go
    Perl: 5.30.3 - /usr/bin/perl
    Python: 3.9.9 - /Users/scamden/.pyenv/shims/python
    Python3: 3.9.9 - /Users/scamden/.pyenv/shims/python3
    Ruby: 2.6.10 - /usr/bin/ruby
    Rust: 1.63.0 - /Users/scamden/.cargo/bin/rustc
  Databases:
    PostgreSQL: 14.8 - /opt/homebrew/opt/postgresql@14/bin/postgres
    SQLite: 3.39.5 - /usr/bin/sqlite3
  Browsers:
    Chrome: 118.0.5993.117
    Safari: 17.0.
```
</details>

**Search keywords**: typescript codemod satisifies data-grid

### PR DESCRIPTION
**Pull request by MBilalShafi**

### CODE REVIEW

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md [hunk 1]
```diff
@@ -59,10 +59,10 @@ You can either run it on a specific file, folder, or your entire codebase when c
 
 ```bash
 // Data Grid specific
-npx @mui/x-codemod v7.0.0/data-grid/preset-safe <path>
+npx @mui/x-codemod@latest v7.0.0/data-grid/preset-safe <path>
```
Comment (MBilalShafi) [inline docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:62]: Context: https://github.com/mui/mui-x/pull/11521#discussion_r1436637173

Comment (LukasTy) [reply docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:62]: Nice. Shouldn't hurt and ensures that people will use the latest package every time. 👍 

Have you considered adding `<!-- #default-branch-switch -->` or `<!-- #default-tag-switch -->` in all of these cases as they would need to change during the pre-release period. 🤔

**Review by LukasTy — APPROVED** [review]

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md (hunk 1 above)
Comment (MBilalShafi) [reply docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:62]: > they would need to change during the pre-release period.

Right, when pre-release for v8 starts, for v8 codemods (or for file `migration-data-grid-v7.md`), we could need to do `npx @mui/x-codemod@next` but the v7 ones would work fine with `npx @mui/x-codemod@latest`, so maybe keeping the `<!-- #default-branch-switch -->` [here](https://github.com/mui/mui-x/blob/783d7edf567e1865a2b4292b4df06d352a1418e5/packages/x-codemod/README.md#L13) should be enough. Wdyt?

Comment (LukasTy) [reply docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:62]: Yup, fair point. There is no real need to switch the tag on old migration docs. 👌 
If you think that it makes sense, we can add it only on `README.md`. 👍

**Pull request merged**